### ISSUE TITLE
Cann't pass test cases locally, but pass on platform

### ISSUE THREAD

**Issue by zy410419243** [circleci/circleci-docs#3040]
# Description
As title

# Location
[This is the report](https://circleci.com/gh/breathing-is-fun/mini-xmind/248)

It shows passed, but you can open the `test` and then you will find it failed.

# Steps to Reproduce
This is the [branch](https://github.com/zy410419243/mini-xmind/commits/circle-problem)  can reproduce this case.

- checkout

- install

- run `npm test`, you will find test fails caused by `DataCollector is not defined`, but [there](https://github.com/zy410419243/mini-xmind/commits/circle-problem), the latest commit passed by mistake

**Comment by teesloane** [comment]
Hi @zy410419243 ! Thanks for filing an issue.

This is an interesting case and I'm not entirely sure why it's happening. But! I have a possible idea. I noticed that you are using your own tool for running a series of web related tasks — `nino-cli`. I took a look at some of its code. 

So, on [this line](https://github.com/breathing-is-fun/nino-cli/blob/master/src/test.js#L14) the handler for a possible `err` is commented out, so I'm guessing it doesn't bubble up as a "failure" with the correct exit code to the parent process. So, in a sense, `npm test` _is succeeding_, but the child process can still fail without telling the parent. So, while it looks like the tests are passing locally, when you run on CircleCI, since it does not see a failure exit code, it considers the step a success. 

Let me know if this is any help, and if not, we'll try and figure out what else the answer could be!
